src/main/java/com/example/bookstoreapp/services/impl/CartItemStateServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.example.bookstoreapp.services.impl;
+
+import com.example.bookstoreapp.entities.CartItemEntity;
+import com.example.bookstoreapp.exceptions.AppRuntimeException;
+import com.example.bookstoreapp.models.*;
+import com.example.bookstoreapp.repositories.CartItemEntityRepository;
+import com.example.bookstoreapp.services.CartItemStateService;
+import com.example.bookstoreapp.services.UserContextService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class CartItemStateServiceImpl implements CartItemStateService {
+    @Autowired
+    private CartItemEntityRepository cartItemEntityRepository;
+    @Autowired
+    private UserContextService userContextService;
+    @Autowired
+    private CartItemServiceImpl cartItemServiceImpl;
+    @Override
+    public CartItemStateUpdate updateStste(Long catalogItemId, CartItemStateUpdate cartItemStateUpdate) {
+        Optional<CartItemEntity> cartItemEntity = cartItemEntityRepository.findById(catalogItemId);
```

Comment:
Let say John and Alex added same item to cart. If John does save later then the item for Alex also go save later as per this logic.

We should do save later for John and Alex individually